src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/ccd/CcdEventProcessor.java

```diff
@@ -38,12 +46,15 @@ public void processMessage(String message) throws JsonProcessingException {
             eventInformation.getCaseTypeId()
         );
 
-        for (CaseEventHandler handler : handlerServices) {
-            List<? extends EvaluateResponse> results = handler.evaluateDmn(eventInformation);
-            if (!results.isEmpty()) {
-                handler.handle(results, eventInformation);
-            }
+        boolean isTaskInitiationEnabled = featureFlagProvider.getBooleanValue(TASK_INITIATION_FEATURE);
+
+        if (isTaskInitiationEnabled) {
+            handlerServices.forEach(handler -> {
+                List<? extends EvaluateResponse> results = handler.evaluateDmn(eventInformation);
+                if (!results.isEmpty()) {
+                    handler.handle(results, eventInformation);
+                }
+            });
         }
```

Comment:
Maybe we should add some logging in here so we can see that messages are being consumed ok from the Topic but not being executed.  Will be good to see that when we are live.